esp-hal/src/spi/master.rs

```diff
@@ -551,6 +551,11 @@ where
             guard: self.guard,
         }
     }
+
+    /// Sends `words` to the slave. Returns the `words` received from the slave
+    pub async fn transfer_async<'w>(&mut self, words: &'w mut [u8]) -> Result<&'w [u8], Error> {
```

Comment:
Maybe I'm missing something but with this PR we have one inherent async fn here - what is special about `transfer` that we add it here but don't add `read_async`, `write_async` etc.?

The naming differs a bit from other places where we have `transfer` vs `transfer_in_place` (i.e. this should be `transfer_in_place_async`)

Reply:
This function is a counterpart of:

```rust
/// Sends `words` to the slave. Returns the `words` received from the slave
pub fn transfer<'w>(&mut self, words: &'w mut [u8]) -> Result<&'w [u8], Error> {
```

I didn't add the others - write_bytes is easy, `write_byte` may be, too, but `read_byte`, I frankly don't understand :)

Reply:
I was just wondering about the other `async fn`s 

rg. naming - we have
`pub async fn transfer_async(&mut self, read: &mut [u8], write: &[u8]) -> Result<(), Error>`
`pub async fn transfer_async_in_place(&mut self, word: &mut [u8]) -> Result<(), Error>`

So probably `pub fn transfer<'w>(&mut self, words: &'w mut [u8]) -> Result<&'w [u8], Error>` is already not well aligned with the async functions' naming

Reply:
IMHO that will already have to be cleaned up by the stabilisation work, async functions will be suffixed with `_async`, but that's not exactly my goal with this PR.